core/src/free/java/de/danoeh/antennapod/core/ClientConfig.java

```diff
@@ -1,6 +1,16 @@
 package de.danoeh.antennapod.core;
 
 import android.content.Context;
+import java.security.Security;
+
+/*
+ * If you get an error here ("package org.conscrypt does not exist"), you are probably doing a free
+ * build and didn't pass "-PfreeBuild" to gradle (e.g. "./gradlew assembleFreeRelease -PfreeBuild").
+ *
+ * If you are doing a non-free build using "assembleRelease" or "assembleDebug" and get this error,
+ * use "assemblePlayRelease" or "assemblePlayDebug" instead (e.g. "./gradlew assemblePlayRelease").
```

Comment:
Could this be fixed by using `freeImplementation "org.conscrypt:conscrypt-android:$conscryptVersion"` instead of using the free build hack? I am not sure if that hack is even needed anymore but that's out of scope for your PR :) I created #4457 for that.

Reply:
Interesting! If you think the "-PfreeBuild" hack can be removed in master (if it's reliable enough for the 2.0.0 release) then I'll be happy to change my PR accordingly.

That reminds me, if it gets changed the circleci goals and F-Droid build meta should then be updated as well. I remember when I first worked on this PR and tried to figure out how to "use the Java preprocessor"... it took some time to find the -PfreeBuild  used for the f-droid meta (but not circleci). That's why I wrote the comment above, telling others to use it (and adding it to circleci). It'd be great if it can be banish altogether.

Now Java just needs a preprocessor so all those "duplicate" Free/Play files can be joined together. :smile:

Reply:
Yeah, that comment is pretty useful. Though, making it "just work" would be even better than a comment :) I think Android Studio automatically selects the free flavour by default, so every new contributor would be suspended by this.

> If you think the "-PfreeBuild" hack can be removed in master

My idea would be to still use `-PfreeBuild` for the proprietary dependencies and to only use `freeImplementation` for conscrypt. As far as I understood the reason for originally adding the hack is that Gradle sometimes builds other flavours in addition to the one you actually need. On F-Droid's build machine without the proprietary libraries, this then causes issues. The other way around (loading conscrypt unneccessarily) does not hurt that much. This still needs to be confirmed but I am pretty sure that when using `freeImplementation`, conscrypt does not end up in the final `play` build. Even if the Gradle error was still there, it would only affect the build and not the final apk.